lib/runner.js

```diff
@@ -52,11 +58,45 @@ class Runner extends Emittery {
 			if (!scheduledStart) {
 				scheduledStart = true;
 				process.nextTick(() => {
+					if (lastMetadata) {
+						const includedInRange = this.lines.some(n => n >= lastCallsiteLineNumber);
+						lastMetadata.exclusive = this.match.length === 0 ?
+							includedInRange :
+							includedInRange && lastMetadata.exclusive;
+					}
+
 					hasStarted = true;
 					this.start();
 				});
 			}
 
+			if (this.lines.length > 0) {
+				this.runOnlyExclusive = true;
+				let callsiteLineNumber;
+
+				for (const jsCallsite of callsites()) {
+					const callsite = sourceMapSupport.wrapCallSite(jsCallsite);
+					const fileName = callsite.getFileName();
+					if (fileName === this.file) {
```

Comment:
I imagine this will break when using AVA with pre-compiled test files. The selected file may be `build/test.js` but `source-map-support` may translate the callsite to `src/test.js`. You should check the files before the source map is applied.

This also needs a comment to explain why we need to check the files in the first place (because the `test()` may be invoked in a helper file — which is its own edge case that I wouldn't know how to deal with).

Reply:
We can't check the file before the source map is applied after the translation line numbers are all changed, (`a.js:5` might end up running line 6.)

Reply:
> We can't check the file before the source map is applied after the translation line numbers are all changed, (`a.js:5` might end up running line 6.)

Yes I get that. My point though is that applying the source map may change the `fileName` of the call site.